include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,326 @@
+
+namespace seqan3
+{
+
+//!\brief Alias for an std::vector of seqan3::cigar.
+//!\ingroup cigar
+using cigar_vector = std::vector<cigar<>>;
+
+/*!\brief Specialized implementation of the stream operator for a container of
+ *        cigar elements; delegates to `cig.to_string()`.
+ * \relates cigar
+ * \tparam range_type Must model seqan3::std::ranges::ForwardRange and its
+ *                    reference type must be seqan3::cigar.
+ * \param  os         The output stream you are printing to.
+ * \param  cigv       The container of cigar elements.
+ * \returns A reference to the output stream.
+ */
+template <std::ranges::ForwardRange range_type>
+//!\cond
+    requires std::Same<reference_t<range_type>, cigar<> &>
+//!\endcond
+std::ostream & operator<<(std::ostream & os, range_type const & cigv)
+{
+    std::for_each(cigv.begin(), cigv.end(), [&os] (auto cig) { os << cig; } );
+    return os;
+}
+
+} // namespace seqan3
+
+// ------------------------------------------------------------------
+// create cigar_vector out of alignment
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+/*!\brief Compares two aligned sequence values and returns their seqan3::cigar_op.
+ * \ingroup aligned_sequence
+ * \tparam reference_value_type Must be equality comparable to seqan3::gap.
+ * \tparam query_value_type     Must be equality comparable to seqan3::gap.
+ * \param  reference_value      Considered indirectly related to the cigar operation.
+ * \param  query_value          The value the cigar operation is based on.
+ * \param  extended_cigar Whether to print the extended cigar alphabet or not. See seqan3::cigar_op.
+ * \returns A seqan3::cigar_op representing the alignment operation between the
+ *          two values.
+ *
+ * \details
+ *
+ * \attention Note that a seqan3::cigar_op, of a cigar element describing an
+ *            alignment at a certain position, is always based on one of the
+ *            two sequences in the pairwise alignment. In this case, the resulting
+ *            operation is based on the \p query value.
+ */
+template<typename reference_value_type, typename query_value_type>
+//!\cond
+    requires std::EqualityComparableWith<std::remove_reference_t<reference_value_type>, gap> &&
+             std::EqualityComparableWith<std::remove_reference_t<query_value_type>, gap>
+//!\endcond
+cigar_op compare_aligned_values(reference_value_type const reference_value,
+                                query_value_type const query_value,
+                                bool const extended_cigar)
+{
+    return (reference_value == gap::GAP)
+        ? (query_value == gap::GAP)
+            ? cigar_op::P
+            : cigar_op::I
+        : (query_value == gap::GAP)
+            ? cigar_op::D
+            : (extended_cigar)
+                ? (query_value == reference_value)
+                    ? cigar_op::EQ
+                    : cigar_op::X
+                : cigar_op::M;
```

Comment:
really?? This nested `?:` is hard to understand and debug. Can we use temporary bool variables to reduce/structure this or just have if statements? Is the efficiency so much better to justify this construct?

Reply:
It's so much shorter than as many if clauses though :open_mouth: 
I don't know about debugging but I think with the indentation it is easy to read.